python_modules/dagster/dagster/core/definitions/run_config.py

```diff
@@ -194,7 +194,9 @@ def get_inputs_field(
     for name, inp in solid.definition.input_dict.items():
         inp_handle = SolidInputHandle(solid, inp)
         has_upstream = input_has_upstream(dependency_structure, inp_handle, solid, name)
-        if asset_layer.asset_key_for_input(handle, name) and not has_upstream:
+        if inp.input_manager_key and has_upstream:
```

Comment:
double check that this condition makes sense and makes sense wrt the rest of the options